bokehjs/src/coffee/models/sources/data_source.ts

```diff
@@ -24,11 +24,19 @@ export abstract class DataSource extends Model {
     this.prototype.type = "DataSource"
 
     this.define({
-      selected: [ p.Any, () => hittest.create_hit_test_result() ], // TODO (bev)
-      callback: [ p.Any                                         ], // TODO: p.Either(p.Instance(Callback), p.Function) ]
+      selected: [ p.Instance                        ], // TODO (bev)
+      callback: [ p.Any                             ], // TODO: p.Either(p.Instance(Callback), p.Function) ]
     })
   }
 
+  initialize(): void {
+    super.initialize()
+
+    if (!this.selected) {
+      this.selected = new Selection()
+    }
```

Comment:
Why not make this the default value for `selected` property?

Reply:
I also wasn't able to set this default on the python side. If I add the default of `lambda: Selection`, I get the following error:

```
Traceback (most recent call last):
  File "selection_model.py", line 1, in <module>
    from bokeh.plotting import figure, show
  File "/Users/claire/Documents/bokeh/bokeh/plotting/__init__.py", line 4, in <module>
    from ..models import ColumnDataSource; ColumnDataSource
  File "/Users/claire/Documents/bokeh/bokeh/models/__init__.py", line 16, in <module>
    from .annotations import *
  File "/Users/claire/Documents/bokeh/bokeh/models/annotations.py", line 16, in <module>
    from ..core.property_mixins import FillProps, LineProps, TextProps
  File "/Users/claire/Documents/bokeh/bokeh/core/property_mixins.py", line 71, in <module>
    class FillProps(HasProps):
  File "/Users/claire/Documents/bokeh/bokeh/core/property_mixins.py", line 79, in FillProps
    fill_alpha = NumberSpec(default=1.0, accept_datetime=False, help=_alpha_help % "fill paths")
  File "/Users/claire/Documents/bokeh/bokeh/core/properties.py", line 1595, in __init__
    super(NumberSpec, self).__init__(key_type, Float, default=default, help=help)
  File "/Users/claire/Documents/bokeh/bokeh/core/properties.py", line 1531, in __init__
    help=help
  File "/Users/claire/Documents/bokeh/bokeh/core/properties.py", line 653, in __init__
    super(Either, self).__init__(default=default, help=help)
  File "/Users/claire/Documents/bokeh/bokeh/core/property/bases.py", line 65, in __init__
    self.validate(self._raw_default())
  File "/Users/claire/Documents/bokeh/bokeh/core/properties.py", line 690, in validate
    if not (value is None or any(param.is_valid(value) for param in self.type_params)):
  File "/Users/claire/Documents/bokeh/bokeh/core/properties.py", line 690, in <genexpr>
    if not (value is None or any(param.is_valid(value) for param in self.type_params)):
  File "/Users/claire/Documents/bokeh/bokeh/core/property/bases.py", line 259, in is_valid
    self.validate(value)
  File "/Users/claire/Documents/bokeh/bokeh/core/properties.py", line 1299, in validate
    raise ValueError("expected an element of %s, got %r" % (self, value))
  File "/Users/claire/Documents/bokeh/bokeh/core/properties.py", line 1279, in __str__
    return "%s(%s, %s)" % (self.__class__.__name__, self.keys_type, self.values_type)
  File "/Users/claire/Documents/bokeh/bokeh/core/properties.py", line 663, in __str__
    return "%s(%s)" % (self.__class__.__name__, ", ".join(map(str, self.type_params)))
  File "/Users/claire/Documents/bokeh/bokeh/core/properties.py", line 435, in __str__
    return "%s(%s)" % (self.__class__.__name__, self.instance_type.__name__)
  File "/Users/claire/Documents/bokeh/bokeh/core/properties.py", line 445, in instance_type
    self._instance_type = getattr(import_module(module, "bokeh"), name)
  File "/anaconda/envs/bokeh/lib/python3.5/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/Users/claire/Documents/bokeh/bokeh/models/transforms.py", line 17, in <module>
    from .sources import ColumnarDataSource
  File "/Users/claire/Documents/bokeh/bokeh/models/sources.py", line 19, in <module>
    class DataSource(Model):
  File "/Users/claire/Documents/bokeh/bokeh/models/sources.py", line 26, in DataSource
    """)
  File "/Users/claire/Documents/bokeh/bokeh/core/properties.py", line 432, in __init__
    super(Instance, self).__init__(default=default, help=help)
  File "/Users/claire/Documents/bokeh/bokeh/core/property/bases.py", line 65, in __init__
    self.validate(self._raw_default())
  File "/Users/claire/Documents/bokeh/bokeh/core/property/bases.py", line 133, in _raw_default
    return self._copy_default(self._default)
  File "/Users/claire/Documents/bokeh/bokeh/core/property/bases.py", line 123, in _copy_default
    return default()
  File "/Users/claire/Documents/bokeh/bokeh/models/sources.py", line 24, in <lambda>
    selected = Instance(Selection, lambda: Selection(), help="""
  File "/Users/claire/Documents/bokeh/bokeh/model.py", line 220, in __init__
    default_theme.apply_to_model(self)
  File "/Users/claire/Documents/bokeh/bokeh/themes/theme.py", line 158, in apply_to_model
    model.apply_theme(self._for_class(model.__class__))
  File "/Users/claire/Documents/bokeh/bokeh/themes/theme.py", line 143, in _for_class
    self._add_glyph_defaults(base, combined)
  File "/Users/claire/Documents/bokeh/bokeh/themes/theme.py", line 126, in _add_glyph_defaults
    from ..models.glyphs import Glyph
  File "/Users/claire/Documents/bokeh/bokeh/models/glyphs.py", line 44, in <module>
    from ..core.property_mixins import FillProps, LineProps, ScalarFillProps, ScalarLineProps, TextProps
ImportError: cannot import name 'FillProps'
```

Reply:
Well, this is something that can be dealt with, but nothing we should be concerned with in this PR.